controller/pptx_parser_/parser.py

```diff
@@ -0,0 +1,36 @@
+from pptx import Presentation
+
+
+def parse_presentation(file_path: str) -> list:
+    """
+    Parse a PowerPoint presentation and extract the text from each slide.
+
+    This function opens a PowerPoint file specified by `file_path`, and for each slide in the presentation, it extracts
+    all the text content and appends it to a list. The lists of slide text are then added to a master list, which is
+    returned.
+
+    Args:
+        file_path (str): The path to the PowerPoint file to parse.
+
+    Returns:
+        list: A list of lists, where each sublist contains the text content of a single slide in the presentation.
+    """
+    presentation = Presentation(file_path)
+    slides = []
+    for slide in presentation.slides:
```

Comment:
4 for loops inside each other is a bit too much, please think of a better way to do so.